src/Fixer/Phpdoc/PhpdocArrayStyleFixer.php

```diff
@@ -0,0 +1,130 @@
+    {
+        return $tokens->isTokenKindFound(T_DOC_COMMENT);
+    }
+
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        $codeSample = <<<'PHP'
+            <?php
+            /**
+             * @param bool[] $a
+             * @param array<int> $b
+             * @param list<int> $c
+             * @param array<string, int> $d
+             */
+
+            PHP;
+
+        return new FixerDefinition(
+            'PHPDoc list types must be written in configured style.',
+            [
+                new CodeSample($codeSample),
+                new CodeSample($codeSample, ['strategy' => self::STRATEGY_FROM_ARRAY_TO_LIST]),
+                new CodeSample($codeSample, ['strategy' => self::STRATEGY_FROM_BRACKETS_TO_LIST]),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * Must run before PhpdocAlignFixer, PhpdocTypesOrderFixer.
+     * Must run after AlignMultilineCommentFixer, CommentToPhpdocFixer, PhpdocIndentFixer, PhpdocScalarFixer, PhpdocToCommentFixer, PhpdocTypesFixer.
+     */
+    public function getPriority(): int
+    {
+        return 1;
+    }
+
+    protected function createConfigurationDefinition(): FixerConfigurationResolverInterface
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('strategy', 'Which part of the conversion - brackets (`[]`) to `array` to `list` - to perform.'))
+                ->setAllowedValues([self::STRATEGY_FROM_ARRAY_TO_LIST, self::STRATEGY_FROM_BRACKETS_TO_ARRAY, self::STRATEGY_FROM_BRACKETS_TO_LIST])
+                ->setDefault(self::STRATEGY_FROM_BRACKETS_TO_ARRAY)
+                ->getOption(),
+        ]);
+    }
+
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
+    {
+        for ($index = $tokens->count() - 1; $index > 0; --$index) {
+            if (!$tokens[$index]->isGivenKind([T_DOC_COMMENT])) {
+                continue;
+            }
+
+            $docBlock = new DocBlock($tokens[$index]->getContent());
+
+            foreach ($docBlock->getAnnotations() as $annotation) {
+                if (!$annotation->supportTypes()) {
+                    continue;
+                }
+
+                $typeExpression = $annotation->getTypeExpression();
+                if (null === $typeExpression) {
+                    continue;
+                }
+
+                $annotation->setTypes([$this->fixType($typeExpression->toString())]);
+            }
+
+            $newContent = $docBlock->getContent();
+            if ($newContent === $tokens[$index]->getContent()) {
+                continue;
+            }
+
+            $tokens[$index] = new Token([T_DOC_COMMENT, $newContent]);
+        }
+    }
+
+    private function fixType(string $type): string
+    {
+        if (self::STRATEGY_FROM_ARRAY_TO_LIST !== $this->configuration['strategy']) {
+            do {
+                $type = Preg::replace('/([\\\\a-zA-Z0-9_>]+)\[\]/', 'array<$1>', $type, -1, $count);
```

Comment:
All replacements must be done on parsed TypeExpression data.

Here are a few examples that will currently either be unfixed or fixed wrongly: https://phpstan.org/r/4350c3a6-cac5-4f26-83d2-0fcd44774fde